core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -107,6 +104,11 @@
     protected String containerId;
     protected String containerName;
 
+    /**
+     * The approach to determine if the container is ready.
+     */
+    protected WaitStrategy waitStrategy;
```

Comment:
Wanted to avoid additional assignment but happy to change if you feel strongly :)